### PR TITLE
feat/wysiwyg editor

### LINKED ISSUE #51: Create simple WYSIWYG editor
Skjermbilde 2020-09-20 kl  15 24 47

Ref:
- https://github.com/ianstormtaylor/slate

### PR DESCRIPTION
**Pull request by TanyaDolgopolova**
Added WYSIWYG editor component 
Screenshot 2020-10-12 at 12 55 55
Close #51

### CODE REVIEW

File: src/components/WYSIWYGEditor/WYSIWYGEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+                        }
+                    }
+                }}
+            />
+        </Slate>
+      </div>
+    )
+};
+
+const toggleBlock = (editor: ReactEditor, format: any) => {
```
Comment (kvokov) [inline src/components/WYSIWYGEditor/WYSIWYGEditor.tsx:64]: Please move all this staff to separate  file like `EditorComponents`

**Review by kvokov — COMMENTED** [review]
Also please check:
- right colors, spaces, etc. in design spec
- indent - 2 spaces ☝️

**Review by kvokov — APPROVED** [review]
👍

File: src/components/WYSIWYGEditor/EditorComponents.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    <div>
+      <link
+        rel="stylesheet"
+        href="https://fonts.googleapis.com/icon?family=Material+Icons"
+      />
+      <span
+        {...props}
+        ref={ref}
+        className={`material-icons ${className ? className : ''}`}
+      />
```
Comment (cobraz) [inline src/components/WYSIWYGEditor/EditorComponents.tsx:65]: Do we need to add Material Icons?

@AnneMatilde: do we have our own icons? Or should we just use Material Design for now, and maybe change it later?

File: src/components/WYSIWYGEditor/EditorComponents.tsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+          ? active
+            ? 'white'
+            : '#aaa'
+          : active
+          ? 'black'
+          : '#ccc',
+      }}
+    />
+  ),
+);
```
Comment (cobraz) [inline src/components/WYSIWYGEditor/EditorComponents.tsx:49]: Do we need a new separate button?

**Review by cobraz — COMMENTED** [review]
Great work! 🙏

I'm wondering if it would be better to spread some of the components into more files. What do you think, @TanyaDolgopolova?

And, maybe we should name it `Editor` instead?

File: src/components/WYSIWYGEditor/EditorComponents.tsx (hunk 2 above)
Comment (TanyaDolgopolova) [reply src/components/WYSIWYGEditor/EditorComponents.tsx:65]: I can add icon files from editor folder to the icons folder, but for that I need svg files

File: src/components/WYSIWYGEditor/EditorComponents.tsx (hunk 3 above)
Comment (TanyaDolgopolova) [reply src/components/WYSIWYGEditor/EditorComponents.tsx:49]: I added this button as we have some different logic comparing with buttons that already exist in the project. I need to change style of the button by the reversed and active props as we need to change button state by the selected part of text

**Comment by TanyaDolgopolova** [comment]
> Great work! 🙏
> 
> I'm wondering if it would be better to spread some of the components into more files. What do you think, @TanyaDolgopolova?
> 
> And, maybe we should name it `Editor` instead?

Yeah, I will see what I can do about spreading components

File: src/components/WYSIWYGEditor/EditorComponents.tsx (hunk 2 above)
Comment (AnneMatilde) [reply src/components/WYSIWYGEditor/EditorComponents.tsx:65]: We have a small icon pack. All icons in svg can be found here: https://drive.google.com/drive/folders/1HTr8f8R1JfQEluwxyPRddMaiZ9vCUrwL?usp=sharing

If there are any icons we're missing, just use Material Design icons 😊

Comment (TanyaDolgopolova) [reply src/components/WYSIWYGEditor/EditorComponents.tsx:65]: Yeah, I have check it out. There is one icon from the menu on this google drive, @cobraz @kvokov should I use only this icon and found another from the internet?

Comment (cobraz) [reply src/components/WYSIWYGEditor/EditorComponents.tsx:65]: I think we can stick with Material Design for now, as is. Later, we can probably adjust and add our own? I talked about it with @AnneMatilde already, so let's resolve this conversation :)

File: src/components/Editor/Editor.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            format="bulleted-list"
+            icon="format_list_bulleted"
+          />
+        </EditorToolbar>
+        <Editable
+          renderElement={renderElement}
+          renderLeaf={renderLeaf}
+          placeholder="Enter some rich text…"
+          spellCheck
+          autoFocus
```
Comment (cobraz) [inline src/components/Editor/Editor.tsx:64]: Can I suggest that these (placeholder, spellCheck, autoFocus) are from props instead?

Comment (cobraz) [reply src/components/Editor/Editor.tsx:64]: We could probably do with a {...props} ?

File: src/components/Editor/Editor.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+} from './components';
+
+const HOTKEYS: { [key: string]: string } = {
+  'mod+b': 'bold',
+  'mod+i': 'italic',
+  'mod+u': 'underline',
+  'mod+`': 'code',
+};
+
+export const Editor = (props: any) => {
```
Comment (cobraz) [inline src/components/Editor/Editor.tsx:26]: Can I suggest that we use Slate's Editor props for this instead of `any`?

**Review by cobraz — APPROVED** [review]
Thank you! 🙏 That looks so much better :D

I left some other nitpicking comments.

File: src/components/Editor/Editor.tsx (hunk 4 above)
Comment (TanyaDolgopolova) [reply src/components/Editor/Editor.tsx:64]: > Can I suggest that these (placeholder, spellCheck, autoFocus) are from props instead?

Okay, I will change it

**Review by cobraz — APPROVED** [review]
LGTM 🎉

**Pull request merged**